app/models/container_node.rb

```diff
@@ -72,17 +73,39 @@ def perf_rollup_parents(interval_name = nil)
     [ext_management_system] unless interval_name == 'realtime'
   end
 
-  def ipaddress
+  def kubernetes_hostname
     labels.find_by(:name => "kubernetes.io/hostname").try(:value)
   end
 
   def cockpit_url
-    URI::HTTP.build(:host => ipaddress, :port => 9090)
+    URI::HTTP.build(:host => kubernetes_hostname, :port => 9090)
   end
 
   def evaluate_alert(_alert_id, _event)
     # currently only EmsEvents from hawkular are tested for node alerts,
     # and these should automaticaly be translated to alerts.
     true
   end
+
+  supports :external_logging_support do
+    unless ext_management_system.respond_to?(:external_logging_route_name)
+      unsupported_reason_add(:external_logging_support, _('This provider type does not support Extrenal Logging'))
+    end
+  end
+
+  def external_logging_query
+    nil # {}.to_query # TODO
+  end
+
+  def external_logging_path
+    fqdn = (kubernetes_hostname || name || "")
+    hostname = fqdn.split('.').first || ""
+    query = "bool:(filter:(or:!("
+    query = "#{query}(term:(hostname:'#{fqdn}'))" if !fqdn.blank?
+    query = "#{query}," if !hostname.blank? && !fqdn.blank?
+    query = "#{query}(term:(hostname:'#{hostname}'))" if !hostname.blank?
+    query = "#{query})))"
+    index = ".operations.*"
```

Comment:
@enoodle mmmm... logic is still incomplete because `fqdn.split('.').first` could generate an hostname that is `== fqdn`:

```ruby
fqdn = "node_short_hostname"
hostname = fqdn.split('.').first || ""
=> "node_short_hostname"
```

Also you check multiple times (at least 4) for `!fqdn.blank?` and `!hostname.blank?`

Since this search is term-based couldn't you just do something like:

```ruby
node_hostnames = [kubernetes_hostname || name]  # node name cannot be empty, it's an ID
node_hostnames.push(node_hostnames.first.split('.').first)
node_hostnames_query = node_hostnames.uniq.map{|x| "(term:(hostname:'#{x}'))"}.join(",")
...
```

Which for `test.example.com` would generate a `node_hostnames_query = "(term:(hostname:'test.example.com')),(term:(hostname:'test'))"`

Also it seems that the URL may deserve a constant.